server/src/main/java/io/crate/types/StringType.java

```diff
@@ -116,6 +117,17 @@ public Query rangeQuery(String field,
                 }
                 return null;
             }
+
+            @Override
+            public Query termsQuery(String field, List<Object> nonNullValues, boolean hasDocValues, boolean isIndexed) {
+                if (isIndexed) {
+                    return new TermInSetQuery(field, nonNullValues.stream().map(BytesRefs::toBytesRef).toArray(BytesRef[]::new));
```

Comment:
The Ctor supporting arrays will convert them back to a list, so we can avoid this roundtrip.
```suggestion
                    return new TermInSetQuery(field, nonNullValues.stream().map(BytesRefs::toBytesRef)..toList());
```